pkgs/stdenv/linux/default.nix

```diff
@@ -1,8 +1,55 @@
 # This file constructs the standard build environment for the
-# Linux/i686 platform.  It's completely pure; that is, it relies on no
+# Linux platform.  It's completely pure; that is, it relies on no
 # external (non-Nix) tools, such as /usr/bin/gcc, and it contains a C
 # compiler and linker that do not search in default locations,
 # ensuring purity of components produced by it.
+#
+# It starts from prebuilt seed bootstrapFiles and creates series of
+# nixpkgs instances (stages) to gradually rebuilt bootstrapFiles
+# with nixpkgs equivalents of packages.
+#
+# Goals of the bootstrap process:
+# 1. final stdenv must not contain any of the bootstrap files
+#    (the only current violation is libgcc_s.so in glibc).
+# 2. final stdenv must not contain any binaries generated directly
+#    by bootstrap compiler (the only currently known violation is
+#    all of glibc).
```

Comment:
```suggestion
# 2. final stdenv must not contain any of the bootstrap files
#    (the only current violations are: libgcc_s.so in glibc,
#    the lib{mpfr,mpc,gmp,isl} which are statically linked
#    into the final gcc).
```

See `bootstrap-stage3`, where we use the bootstrap compiler is used to compile `.a` versions of those four libraries.  This drives me nuts, by the way.  I don't know why, but it bothers me a lot.  The code has been there since cfde88976ba4cddd01b1bb28b40afd12ea93a11d.

`nix-tree --derivation $(nix-instantiate -A gcc)` is helpful for untangling all of this.

Reply:
Good point on those static libs!

Reply:
> Good point on those static libs!

I remembered why this bothers me so much: https://github.com/NixOS/nixpkgs/pull/170215/commits/518353b4b616989dd440cae20298502f245accc9